### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by maksym-nemera**

### CODE REVIEW

File: src/sortStudents.ts [hunk 1]
```diff
@@ -1,16 +1,55 @@
...
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudents: Student[] = [...students];
+
+  return sortedStudents.sort((a, b) => {
+    switch (sortBy) {
+      case SortType.AverageGrade:
+        return order === 'asc'
+          ? a.grades
```
Comment (33Orange) [inline src/sortStudents.ts:30]: Extract the average grade calculation into a separate function to improve code readability and maintainability.

File: src/sortStudents.ts [hunk 2]
```diff
@@ -1,16 +1,55 @@
...
+          : Number(b[sortBy]) - Number(a[sortBy]);
+
+      case SortType.Name:
+      case SortType.Surname:
+        return order === 'asc'
+          ? a[sortBy].localeCompare(b[sortBy])
+          : b[sortBy].localeCompare(a[sortBy]);
 
+      default:
+        break;
```
Comment (33Orange) [inline src/sortStudents.ts:50]: Remove the unnecessary 'break' statement in the default case.
It's better to add error handling if your sorting is in the default case.

File: src/sortStudents.ts [hunk 3]
```diff
@@ -1,16 +1,55 @@
...
+    }
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+    return 0;
```
Comment (33Orange) [inline src/sortStudents.ts:53]: Remove the unnecessary 'return 0;' line as the function already returns the sorted array.

**Review by 33Orange — CHANGES_REQUESTED** [review]
Almost done!

A few moments to fix.

File: src/sortStudents.ts [hunk 4]
```diff
@@ -1,16 +1,54 @@
...
-// create SortOrder type
-export type SortOrder;
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudents: Student[] = [...students];
+
+  return sortedStudents.sort((a, b) => {
```
Comment (dinaminahi) [inline src/sortStudents.ts:31]: It is better to create more descriptive names - a and b tell nothing 
Something like student1 and student2 or firstStudent and secondStudent will be better

**Review by dinaminahi — APPROVED** [review]
Well done 💪

**Pull request closed without merging**